src/mlpack/methods/adaboost/adaboost_regressor.hpp

```diff
@@ -0,0 +1,276 @@
+   * @param minimumGainSplit Minimum gain for splitting a decision tree node.
+   * @param maximumDepth Maximum depth for the tree.
+   * @param dimensionSelector Instantiated dimension selection policy.
+   * @return The average entropy of all the decision trees trained under forest.
+   */
+  template<typename MatType, typename ResponsesType>
+  double Train(const MatType& data,
+               const arma::Row<ResponsesType>& responses,
+               const size_t numTrees = 20,
+               const size_t minimumLeafSize = 10,
+               const double minimumGainSplit = 1e-7,
+               const size_t maximumDepth = 4,
+               DimensionSelectionType dimensionSelector =
+                   DimensionSelectionType());
+
+  /**
+   * Train the AdaBoost regressor on the given data and responses with the given
+   * dataset info and the given number of trees.  The minimumLeafSize parameter
+   * is given to each individual decision tree during tree building.
+   * Optionally, you may specify a DimensionSelectionType to set parameters for
+   * the strategy used to choose dimensions.
+   * This overload can be used to train on categorical data.
+   *
+   * @param data Dataset to train on.
+   * @param datasetInfo Dimension info for the dataset.
+   * @param responses Responses for each training point.
+   * @param numTrees Number of trees in the forest.
+   * @param minimumLeafSize Minimum number of points in each tree's leaf nodes.
+   * @param minimumGainSplit Minimum gain for splitting a decision tree node.
+   * @param maximumDepth Maximum depth for the tree.
+   * @param dimensionSelector Instantiated dimension selection policy.
+   * @return The average entropy of all the decision trees trained under forest.
+   */
+  template<typename MatType, typename ResponsesType>
+  double Train(const MatType& data,
+               const data::DatasetInfo& datasetInfo,
+               const arma::Row<ResponsesType>& responses,
+               const size_t numTrees = 20,
+               const size_t minimumLeafSize = 10,
+               const double minimumGainSplit = 1e-7,
+               const size_t maximumDepth = 4,
+               DimensionSelectionType dimensionSelector =
+                   DimensionSelectionType());
+
+  /**
+   * Predict the class of the given point. If the AdaBoost regressor has not been
+   * trained, this will throw an exception.
+   *
+   * @param point Point to be classified.
+   */
+  template<typename VecType>
+  double Predict(const VecType& point) const;
+
+  /**
+   * Predict the classes of each point in the given dataset.  If the AdaBoost
+   * regressor has not been trained, this will throw an exception.
+   *
+   * @param data Set of points to predict.
+   * @param predictions Output predictions for each point in the dataset.
+   */
+  template<typename MatType, typename ResponsesType>
+  void Predict(const MatType& data,
+                arma::Row<ResponsesType>& predictions) const;
```

Comment:
`arma::Row` should take the `ElemType` and not the `ResponseType`, 
I think you are looking for something like this 
`ResopnsesType& predictions`